src/packets.toit

```diff
@@ -5,12 +5,38 @@
 import bytes
 import binary
 import reader
+import .client show Client  // For toitdoc reference.
 
 import .topic_filter
 
 interface PacketIDAck:
   packet_id -> int
 
+class LastWillConfig:
+  retain/bool
+  qos/int
+  topic/string
+  payload/ByteArray
+
+
+  /**
+  Constructs the configuration of a last-will message.
+
+  The parameters $topic, $payload, $qos and $retain have the same
+    meaning as for $Client.publish, and are used when the last-will message
+    is eventually sent.
+  */
+  constructor .topic .payload --.qos --.retain=false:
+    if not 0 <= qos <= 2: throw "INVALID_QOS"
+
+  /**
+  Variant of $LastWillConfig.constructor.
+  Takes a string ($message) as payload, instead of a ByteArray.
+  */
+  constructor.from_string .topic message/string --.qos --.retain=false:
```

Comment:
Not sure 'from_string' is great here. It is just the message that is from a string, not the full config.

Reply:
Removed it.
The `publish` function only takes byte-arrays too.

Hopefully with the new io library things should get nicer.
done.